### PR TITLE
Sign2.5/Samples

### PR DESCRIPTION
**Pull request by jakubuid**

### CODE REVIEW

File: protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/responses/OnSessionAuthenticateResponseUseCase.kt [hunk 1]
```diff
@@ -60,8 +64,7 @@ internal class OnSessionAuthenticateResponseUseCase(
 
             val pairingTopic = jsonRpcHistoryEntry.topic
             if (!pairingInterface.getPairings().any { pairing -> pairing.topic == pairingTopic.value }) return@supervisorScope //todo: emit error
-//            todo: handle pending session authenticate requests
-//            pairingTopicToResponseTopicMap.remove(pairingTopic)
+            scope.launch { authenticateResponseTopicRepository.delete(pairingTopic.value) }
```
Comment (Elyniss) [inline protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/responses/OnSessionAuthenticateResponseUseCase.kt:67]: why is this being launched in scope? Does it need to be async? If so we might want to add runCatching, otherwise we wont know if something went bad. If no need for async then exception will be caught at the bottom, so all good

File: protocol/sign/src/main/sqldelight/com/walletconnect/sign/storage/data/dao/authenticate_reponse/AuthenticateResponseTopicDao.sq [hunk 2]
```diff
@@ -0,0 +1,17 @@
+CREATE TABLE AuthenticateResponseTopicDao (
```
Comment (Elyniss) [inline protocol/sign/src/main/sqldelight/com/walletconnect/sign/storage/data/dao/authenticate_reponse/AuthenticateResponseTopicDao.sq:1]: I would remove Dao prefix, as this is table

File: sample/wallet/src/main/kotlin/com/walletconnect/sample/wallet/ui/routes/dialog_routes/session_authenticate/SessionAuthenticateRoute.kt [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        itemsIndexed(authRequestUI.messages) { _, item ->
+            Spacer(modifier = Modifier.height(12.dp))
+            Text(
+                text = item.second,
+                modifier = Modifier.padding(vertical = 10.dp, horizontal = 13.dp),
+                style = TextStyle(fontWeight = FontWeight.Medium, fontSize = 13.sp, color = themedColor(darkColor = Color(0xFF9ea9a9), lightColor = Color(0xFF788686)))
+            )
+        }
+    }
+}
```
Comment (Elyniss) [inline sample/wallet/src/main/kotlin/com/walletconnect/sample/wallet/ui/routes/dialog_routes/session_authenticate/SessionAuthenticateRoute.kt:224]: Could we make different messages stand out a little more, like so?

```kotlin
@Composable
fun Messages(authRequestUI: SessionAuthenticateUI) {
    LazyColumn(
        modifier = Modifier
            .padding(20.dp)
            .clip(shape = RoundedCornerShape(25.dp))
            .fillMaxWidth()
            .height(300.dp)
    ) {
        itemsIndexed(authRequestUI.messages) { _, item ->
            Spacer(modifier = Modifier.height(12.dp))
            Column(modifier = Modifier
                .clip(shape = RoundedCornerShape(25.dp))
                .fillMaxWidth()
                .background(themedColor(darkColor = Color(0xFFE4E4E7).copy(alpha = .12f), lightColor = Color(0xFF505059).copy(.1f)))
            ) {
                Text(
                    text = item.second,
                    modifier = Modifier.padding(vertical = 10.dp, horizontal = 13.dp),
                    style = TextStyle(fontWeight = FontWeight.Medium, fontSize = 13.sp, color = themedColor(darkColor = Color(0xFF9ea9a9), lightColor = Color(0xFF788686)))
                )
            }

        }
    }
}
```

Comment (Elyniss) [reply sample/wallet/src/main/kotlin/com/walletconnect/sample/wallet/ui/routes/dialog_routes/session_authenticate/SessionAuthenticateRoute.kt:224]: Screenshot 2024-02-09 at 11 13 18

File: protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/responses/OnSessionAuthenticateResponseUseCase.kt (hunk 1 above)
Comment (jakubuid) [reply protocol/sign/src/main/kotlin/com/walletconnect/sign/engine/use_case/responses/OnSessionAuthenticateResponseUseCase.kt:67]: Yeah I think we need async here. It's DB operation.

File: protocol/sign/src/main/sqldelight/com/walletconnect/sign/storage/data/dao/authenticate_reponse/AuthenticateResponseTopicDao.sq (hunk 2 above)
Comment (jakubuid) [reply protocol/sign/src/main/sqldelight/com/walletconnect/sign/storage/data/dao/authenticate_reponse/AuthenticateResponseTopicDao.sq:1]: All our tables in Sign has the Dao suffix so I don't want to break the convention

File: sample/wallet/src/main/kotlin/com/walletconnect/sample/wallet/ui/routes/dialog_routes/session_authenticate/SessionAuthenticateRoute.kt (hunk 3 above)
Comment (jakubuid) [reply sample/wallet/src/main/kotlin/com/walletconnect/sample/wallet/ui/routes/dialog_routes/session_authenticate/SessionAuthenticateRoute.kt:224]: This one will most likely change, so we're working on the web3wallet team to have one message to sign for multiple chains. I'll come back to this once we have a solution.

File: protocol/sign/src/main/sqldelight/com/walletconnect/sign/storage/data/dao/authenticate_reponse/AuthenticateResponseTopicDao.sq [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+   id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL,
+   pairingTopic TEXT NOT NULL,
+   responseTopic TEXT NOT NULL
```
Comment (TalhaAli00) [inline protocol/sign/src/main/sqldelight/com/walletconnect/sign/storage/data/dao/authenticate_reponse/AuthenticateResponseTopicDao.sq:4]: should either of these be unique or should we have have a unique constraint for both fields together?

**Review by TalhaAli00 — APPROVED** [review]

File: protocol/sign/src/main/sqldelight/com/walletconnect/sign/storage/data/dao/authenticate_reponse/AuthenticateResponseTopicDao.sq (hunk 4 above)
Comment (jakubuid) [reply protocol/sign/src/main/sqldelight/com/walletconnect/sign/storage/data/dao/authenticate_reponse/AuthenticateResponseTopicDao.sq:4]: why do you think they should be unique?
